src/features/broker-protection/actions/extract.js

```diff
@@ -134,8 +134,9 @@ export function createProfile (elementFactory, extractData) {
             // Note: This can return a string, string[], or null
             const extractedValue = extractValue(key, value, evaluatedValue)
 
-            // try to use the extracted value first, then the originally evaluated, falling back to null
-            output[key] = extractedValue || evaluatedValue || null
+            // try to use the extracted value, or fall back to null
+            // this allows 'extractValue' to return null|undefined
+            output[key] = extractedValue || null
```

Comment:
I looked at all the return paths from `extractValue()` - we should respect when it returns `null` or `undefined`. The previous logic here does not do that - because if `extractedValue` is null, it ends up returning the original string.

In my testing, this is *never* the desired outcome.

This change allows the extractor methods to be in control of whether a value will be used, or not.